Using service yaml value as source for the Riemann event service value

dgeorgievski opened the pull request:
I would like to be able to set an arbitrary name for the service setting to avoid some issues originating in special characters used for JMX object names, like commas, white spaces etc

In my case I am using the following pipe for collecting JMX metrics
riemann-jmx -> riemann -> cyanite (Graphite in Clojure)

Visualizations are created with Grafana, which has issues to query metrics if service values are created from object names.

Example, a yaml fragment: 
queries :
-   service     : "hbase.region.server"
  use_service : true
  obj         : "Hadoop:name=RegionServer,service=HBase,sub=Server"

With use_service set to false, or no, the path of a metric stored by Cyanite looks like this
  my_event_host.Hadoop:name=RegionServer,service=HBase,sub=Server.regionCount

with use_service set to true or yes, the metric path becomes
  my_event_host.hbase.region.server.regionCount

which is easier to handle when quering metrics from tools like Grafana.

dgrnbrg commented:
I think this is a fantastic enhancement. The only change I'd like is to make it so that if you provide `service` then it uses that, so that there's no need for the extra flag setting (since you'll only set `service` if you want to change it).

Also, could you add an example (with explanatory comment) to the example config?

That's all we need to merge!

dgeorgievski commented:
Actually that was my first implementation, but then I thought this change could have created unwanted behavior with users that have the service yaml field set, but still expect to see the obj name in the riemann event service field.  I'm glad you are in favor of using the service field for this purpose.

Hopefully, the comments (explanation and examples) for the service field in the example yaml file are not too long.

And lastly, the change was tested successfully against a Riemann 0.2.10 instance.

dgrnbrg commented:
Could you just squash these commits? Then I can merge them. I think that if you run `git reset --soft HEAD~3 && git commit` you'll be able to write the new commit message and have them squashed. Thanks!

Commit:
- Dropped the service flag in favor of using service field when composing Riemann events.
    - Added a description on the role of the service field, and examples on its usage in the example config file.
    - Added a line in .gitignore to ignore *.swp Vim files.

dgeorgievski commented:
Older commits squashed.

dgrnbrg commented:
Thank you for this great contribution!
